### PR TITLE
feat: Implement rename credential change

### PR DESCRIPTION
**Pull request by talwinder50**
partof #1516

- Regarding validation as discussed with UX team, As long as there is a way for the user to differentiate credentials we dont have a problem. Either by different look, or by different name. Therefore removing same name validation for now as discussed with the team

Screen Shot 2022-03-21 at 7 14 54 PM
Screen Shot 2022-03-21 at 7 15 29 PM
Screen Shot 2022-03-21 at 7 15 39 PM

Signed-off-by: talwinder50 <talwinderkaur50@gmail.com>

https://user-images.githubusercontent.com/7862595/159482503-3647db06-06d7-4dcb-bff5-da24f29bed3c.mov

Updated video

https://user-images.githubusercontent.com/7862595/159603351-dce1f619-c57e-4de4-a37a-b62e091cda15.mp4

- Fixed -> - Fixed when the modal is opened it should show the existing name as a placeholder text. 
Screen Shot 2022-03-22 at 8 52 08 PM
- When entering the new valid name the bottom border of the input field is not red now 
Screen Shot 2022-03-22 at 8 52 20 PM
Screen Shot 2022-03-22 at 8 52 08 PM
Screen Shot 2022-03-22 at 8 52 20 PM
Screen Shot 2022-03-22 at 8 52 28 PM
Screen Shot 2022-03-22 at 8 52 44 PM
Screen Shot 2022-03-22 at 8 52 53 PM
Screen Shot 2022-03-22 at 8 53 00 PM
Screen Shot 2022-03-22 at 8 53 11 PM

- 
Screen Shot 2022-03-22 at 8 27 14 PM

Screen Shot 2022-03-22 at 9 28 42 PM

### CODE REVIEW

File: cmd/wallet-web/src/components/CredentialDetails/RenameCredentialModal.vue [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+          });
+
+          let metadata = await credentialManager.getCredentialMetadata(token, credentialId);
+          credentialMutations.setCredentialsOutdated(true);
+          this.showModal = false;
+          this.loading = false;
+          this.submitted = false;
+          this.$emit('close');
+        } catch (e) {
+          console.error('Error while renaming credential:', e);
```
Comment (sudeshrshetty) [inline cmd/wallet-web/src/components/CredentialDetails/RenameCredentialModal.vue:166]: Can't we use standard error handler/display of system error in this case?

File: cmd/wallet-web/src/components/CredentialDetails/RenameCredentialModal.vue [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        const { contents } = await collectionManager.getAll(token);
+        this.vaults = Object.values(contents).map((vault) => vault);
+        const vaultID = this.vaults.find((vault) => vault.name === vaultName).id;
+        try {
+          await credentialManager.updateCredentialMetadata(token, credentialId, {
+            name: this.credentialName,
+            collection: vaultID,
+          });
+
+          let metadata = await credentialManager.getCredentialMetadata(token, credentialId);
```
Comment (sudeshrshetty) [inline cmd/wallet-web/src/components/CredentialDetails/RenameCredentialModal.vue:159]: dead object `metadata`, not used anywhere

**Review by sudeshrshetty — CHANGES_REQUESTED** [review]

File: cmd/wallet-web/src/components/CredentialDetails/RenameCredentialModal.vue (hunk 1 above)
Comment (talwinder50) [reply cmd/wallet-web/src/components/CredentialDetails/RenameCredentialModal.vue:166]: can be done as part of #1600

**Comment by talwinder50** [comment]
> @talwinder50 Why does the recording show "missing credential" page at 1:34 ?

it was some lag in my local i havent seen that in subsequent runs

**Comment by birtony** [comment]
@talwinder50 could you also rebase your PR on top of main, please? There might be some conflicts

File: cmd/wallet-web/src/components/CredentialDetails/RenameCredentialModal.vue (hunk 1 above)
Comment (talwinder50) [reply cmd/wallet-web/src/components/CredentialDetails/RenameCredentialModal.vue:164]: Having further discussion with the design team, system errors in the modals should a toast errors like this https://app.abstract.com/projects/9ec92568-60a2-4fee-8215-5aa997d52428/branches/dc10bcee-d9a1-44f9-b644-8f843c6f1a19/commits/a0222a405989d1245b0548d775b3a9aae6fba3ed/files/89ee2d9d-cd69-4357-9433-e0be2f7748fb/layers/EC425024-0893-4172-98B4-77CDB5298D58


This should be separate task #1600  @sudeshrshetty

File: cmd/wallet-web/src/pages/CredentialDetails.vue [hunk 3]
```diff
@@ -101,7 +115,17 @@ import Flyout from '@/components/Flyout/Flyout.vue';
...
 import FlyoutButton from '@/components/Flyout/FlyoutButton.vue';
 import DeleteCredential from '@/components/CredentialDetails/DeleteCredentialModal.vue';
+import RenameCredential from '@/components/CredentialDetails/RenameCredentialModal.vue';
 
+export const credentialStore = reactive({
+  credentialsOutdated: false,
+});
+
+export const credentialMutations = {
+  setCredentialsOutdated(value) {
```
Comment (birtony) [inline cmd/wallet-web/src/pages/CredentialDetails.vue:125]: And this too

File: cmd/wallet-web/src/components/CredentialDetails/RenameCredentialModal.vue (hunk 1 above)
Comment (sudeshrshetty) [reply cmd/wallet-web/src/components/CredentialDetails/RenameCredentialModal.vue:164]: why cant we show it like existing error scenario for this page? just different text (copied from system error page), add a TODO and fix it later once you get something from design team

Comment (talwinder50) [reply cmd/wallet-web/src/components/CredentialDetails/RenameCredentialModal.vue:179]: I had discussion with design team and have added this as toast message

File: cmd/wallet-web/src/pages/CredentialDetails.vue [hunk 4]
```diff
@@ -101,7 +115,17 @@ import Flyout from '@/components/Flyout/Flyout.vue';
 import FlyoutMenu from '@/components/Flyout/FlyoutMenu.vue';
 import FlyoutButton from '@/components/Flyout/FlyoutButton.vue';
 import DeleteCredential from '@/components/CredentialDetails/DeleteCredentialModal.vue';
+import RenameCredential from '@/components/CredentialDetails/RenameCredentialModal.vue';
 
+export const credentialStore = reactive({
+  credentialsOutdated: false,
```
[the remark this answers is not in the document]
Comment (talwinder50) [reply cmd/wallet-web/src/pages/CredentialDetails.vue:122]: Can you please elaborate , what you mean by this

File: cmd/wallet-web/src/components/CredentialDetails/RenameCredentialModal.vue (hunk 1 above)
Comment (talwinder50) [reply cmd/wallet-web/src/components/CredentialDetails/RenameCredentialModal.vue:179]: Screen Shot 2022-03-22 at 8 27 14 PM
As per the discussion with the design team i have added toast and have requested team to provide abstract design for the same as well

**Review by sudeshrshetty — APPROVED** [review]

File: cmd/wallet-web/src/pages/WACIShare.vue [hunk 5]
```diff
@@ -264,13 +264,12 @@ export default {
...
-            this.token,
-            credential.id
-          );
+          const { id, name, issuanceDate, resolved } =
+            await this.credentialManager.getCredentialMetadata(this.token, credential.id);
           // TODO: issue1410 - add logic to retrieve the list of vaults in which the credential is stored
           const vaultName = 'Unavailable';
-          this.processedCredentials.push({ id, issuanceDate, ...resolved[0], vaultName });
+          this.processedCredentials.push({ id, name, issuanceDate, ...resolved[0], vaultName });
+          console.log('procesed cred', this.processedCredentials, this.processedCredentials.length);
```
Comment (birtony) [inline cmd/wallet-web/src/pages/WACIShare.vue:272]: We need to remove this console log

File: cmd/wallet-web/src/components/CredentialDetails/RenameCredentialModal.vue [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+
+<template>
+  <modal :show="showModal" :show-close-button="true" @close="handleClose">
+    <template #errorToast>
+      <toast-notification
+        v-if="systemError"
+        :title="t('CredentialDetails.RenameModal.errorToast.title')"
+        :description="t('CredentialDetails.RenameModal.errorToast.description')"
+        type="error"
+      ></toast-notification>
```
Comment (birtony) [inline cmd/wallet-web/src/components/CredentialDetails/RenameCredentialModal.vue:15]: We could make this shorter:
```js
      <toast-notification
        v-if="systemError"
        :title="t('CredentialDetails.RenameModal.errorToast.title')"
        :description="t('CredentialDetails.RenameModal.errorToast.description')"
        type="error"
      />
```

**Review by birtony — APPROVED** [review]
Looks good now!

**Pull request merged**